src/Moq/MethodCall.cs

```diff
@@ -469,40 +465,54 @@ public override void RespondTo(Invocation invocation)
 		private sealed class RaiseEventResponse
 		{
 			private Mock mock;
-			private EventInfo @event;
+			private LambdaExpression expression;
 			private Delegate eventArgsFunc;
 			private object[] eventArgsParams;
 
-			public RaiseEventResponse(Mock mock, EventInfo @event, Delegate eventArgsFunc, object[] eventArgsParams)
+			public RaiseEventResponse(Mock mock, LambdaExpression expression, Delegate eventArgsFunc, object[] eventArgsParams)
 			{
 				Debug.Assert(mock != null);
-				Debug.Assert(@event != null);
+				Debug.Assert(expression != null);
 				Debug.Assert(eventArgsFunc != null ^ eventArgsParams != null);
 
 				this.mock = mock;
-				this.@event = @event;
+				this.expression = expression;
 				this.eventArgsFunc = eventArgsFunc;
 				this.eventArgsParams = eventArgsParams;
 			}
 
 			public void RespondTo(Invocation invocation)
 			{
+				object[] args;
+
 				if (this.eventArgsParams != null)
 				{
-					this.mock.DoRaise(this.@event, this.eventArgsParams);
+					args = this.eventArgsParams;
 				}
 				else
 				{
 					var argsFuncType = this.eventArgsFunc.GetType();
 					if (argsFuncType.GetTypeInfo().IsGenericType && argsFuncType.GetGenericArguments().Length == 1)
 					{
-						this.mock.DoRaise(this.@event, (EventArgs)this.eventArgsFunc.InvokePreserveStack());
+						args = new object[] { this.mock.Object, this.eventArgsFunc.InvokePreserveStack() };
 					}
 					else
 					{
-						this.mock.DoRaise(this.@event, (EventArgs)this.eventArgsFunc.InvokePreserveStack(invocation.Arguments));
+						args = new object[] { this.mock.Object, this.eventArgsFunc.InvokePreserveStack(invocation.Arguments) };
 					}
+
+					Mock.RaiseEvent(
+						this.mock,
+						this.expression,
+						this.expression.Split(),
+						args);
 				}
+
+				Mock.RaiseEvent(
+					this.mock,
+					this.expression,
+					this.expression.Split(),
+					args);
```

Comment:
Oops. Some obvious repetition here. Remove the inner call, and put the second one on a single line for brevity.